app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb

```diff
@@ -235,27 +249,29 @@ def add_instance_disk(disks, size, name, location)
         super(disks, size, name, location, "azure")
       end
 
-      def populate_hardware_hash_with_networks(hardware_networks_array, instance)
-        nics = instance.fetch_path('properties', 'networkProfile', 'networkInterfaces')
-
-        nics.each do |n|
-          n['properties'].each do |n_prop|
-            private_network = {
-              :ipaddress => n_prop['properties']['privateIPAddress'],
-              :hostname  => n_prop['name']
-            }.delete_nils
-
-            public_network = {
-              :ipaddress => n_prop['properties']['publicIPAddress'],
-              :hostname  => n_prop['name']
-            }.delete_nils
-
-            hardware_networks_array <<
-              private_network.merge(:description => "private") unless private_network.blank?
-            hardware_networks_array <<
-              public_network.merge(:description => "public") unless public_network.blank?
+      def populate_hardware_hash_with_networks(networks_array, instance)
+        instance.properties.network_profile.network_interfaces.each do |nic|
```

Comment:
Rewrote the whole implementation. The armrest gem no longer queries network ip addresses and inserts to the vm object. Make explicit calls here to get them.